### DISCUSSION TITLE
How to append items to an AsyncNotifier?

### DISCUSSION THREAD

**Discussion by snapsl** [rrousselGit/riverpod#2073] [Q&A]
Hi,
Is there an option to append data to an AsyncNotifier? Usually, I would use the spread operator here.

```dart
@riverpod
class DataList extends _$DataList {
  @override
  FutureOr<List<Data>> build() {
    return [];
  }

  Future<void> getData() async {
    List<Data> data = await fetchData(); 

    // state = [...state, ...data];
  }
}
```
An option would be to access the value, but is this intended?

```dart
state = AsyncValue.data([...state.asData?.value ?? [], ...data]);
```

**Comment by rrousselGit** [comment]
You need to handle the possible error/loading state, hence the current syntax.

There's no right or wrong here. Your code technically works but currently silences errors or loading states (if any). If that's with you, that's alright.


But your `build` method is weird. Chances are `getData` should be invoked in `build` instead.
